File: charts/hermes-agent/files/health.py
# SPDX-License-Identifier: Apache-2.0
"""Readiness is authenticated JSON state, never just an HTTP 200 response."""
import json
import os
import sys
import urllib.request

try:
    request = urllib.request.Request('http://127.0.0.1:8642/health/detailed',
                                     headers={'Authorization': 'Bearer ' + os.environ['API_SERVER_KEY']})
    with urllib.request.urlopen(request, timeout=3) as response:
        healthy = json.load(response).get('readiness', {}).get('status') == 'ok'
    sys.exit(0 if healthy else 1)
except Exception:
    sys.exit(1)

File: charts/openclaw/files/health.mjs
// SPDX-License-Identifier: Apache-2.0
const live=process.argv[2]==='live';
try{
 const response=await fetch('http://127.0.0.1:18789/'+(live?'healthz':'startupz'),{signal:AbortSignal.timeout(3000)});
 if(!response.ok)process.exit(1);
 const body=await response.json();
 process.exit(body.ok===true && body.status===(live?'live':'started')?0:1);
}catch{process.exit(1);}
